File: ai-resume-portal/frontend/src/types/job.ts
export type JobType = 'FULL_TIME' | 'PART_TIME' | 'CONTRACT' | 'INTERNSHIP'

export interface Job {
  id: number
  title: string
  description: string
  requiredSkills: string
  experienceRequired?: string | null
  salaryRange?: string | null
  location: string
  jobType: JobType
  isActive: boolean
  deadline?: string | null
  employerId?: number | null
  employerName?: string | null
  employerCompany?: string | null
  createdAt?: string | null
  updatedAt?: string | null
}

export interface JobRequest {
  title: string
  description: string
  requiredSkills: string
  experienceRequired?: string
  salaryRange?: string
  location: string
  jobType: JobType
  deadline?: string
}

File: ai-resume-portal/backend/src/main/java/com/portal/job/entity/Job.java
package com.portal.job.entity;

import com.portal.user.entity.User;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import java.time.LocalDate;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

@Entity
@Table(name = "jobs")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Job {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "employer_id", nullable = false)
    private User employer;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(nullable = false, columnDefinition = "TEXT")
    private String description;

    @Column(name = "required_skills", nullable = false, columnDefinition = "TEXT")
    private String requiredSkills;

    @Column(name = "experience_required", length = 50)
    private String experienceRequired;

    @Column(name = "salary_range", length = 50)
    private String salaryRange;

    @Column(nullable = false, length = 100)
    private String location;

    @Enumerated(EnumType.STRING)
    @Column(name = "job_type", nullable = false, length = 20)
    private JobType jobType;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;

    @Column
    private LocalDate deadline;

    @CreationTimestamp
    @Column(name = "created_at", updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(name = "updated_at")
    private LocalDateTime updatedAt;
}
